### ISSUE TITLE
Mechanisms for defaulting magit to emacs state instead of evil no longer works on develop branch

### ISSUE THREAD

**Issue by mgsloan** [syl20bnr/spacemacs#10528]
Sorry if this is a duplicate of another issue.  I poked around a bit but didn't see anything for this.

I expect `n` and `p` to go to the next and previous item in magit.  On develop, I couldn't figure out keybindings that did this properly - I tried j / k and M-n and M-p, but those do not have the same behavior of moving between subsections.

Also, I expect `g` to refresh magit, but instead it now wanted me to press `g r`.

I realize that sometimes keybindings need to change, but these are very much so embedded in my muscle memory and switching it and many other magit keybindings around would be a major pain.  Is there some sort of configuration option for this? 

For now I am going to stick with `master`.

**Comment by duianto** [comment]
It seems to be working as you expect, with the `Emacs` editing style on the latest develop branch.

- `n` and `p` navigates to the next and previous line
- `M-n` and `M-p` navigates between section siblings, "Head:", "Recent commits" (and between it's listed commits)
- `g` refreshes the buffer

What you describe seems to be the `Vim` bindings.

- `j` and `k` = the next and previous line
- `[`, `gj` or `M-n` = next sibling
- `]`, `gk` or `M-p` = previous sibling
- `gr` refreshes the buffer

#### System Info :computer:
- OS: windows-nt
- Emacs: 25.3.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 0fa3658c)
- Graphic display: t
- Distribution: spacemacs
- Editing style: emacs
- Completion: helm
- Layers:
```elisp
(autohotkey helm auto-completion emacs-lisp git html org spell-checking syntax-checking treemacs version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS MODULES

**Comment by nixmaniack** [comment]
It would be helpful if you can follow [these guidelines](https://github.com/syl20bnr/spacemacs/wiki/Debugging#how-to-make-a-great-bug-report) for issue creation.

The behaviour could be because of introduction of `evil-magit` or configuration of `evil-magit` or editing style (`vim`, `hybrid` or `emacs`) in use or their configuration.

**Comment by mgsloan** [comment]
I finally got around to revisiting updating my spacemacs to the latest version of the `develop` branch and so I'm revisiting this issue.  Sorry for not following the issue creation guidelines.

There does seem to be an issue.  Along with `dotspacemacs-editing-style 'vim`, my configuration has

```
  (push '("*magit" . emacs) evil-buffer-regexps)
```

as suggested here - https://github.com/syl20bnr/spacemacs/blob/master/doc/FAQ.org#disable-evilification-of-a-mode .  So, the issue is that this mechanism for de-evil-ifying magit no longer works.  I also tried the suggestion that only de-evil-ifies magit-status:

```
  (evil-set-initial-state 'magit-status-mode 'emacs)
```

This also doesn't work.

**Comment by mgsloan** [comment]
Ah, another twist here is that for a long time I've had a keybinding that shadows the usual C-z for `evil-emacs-state` / `evil-exit-emacs-state`.  So, this is why I'm not used to switching the modes.  So, seems I just need to get `evil-emacs-state` invoked initially for magit buffers.

For reference, this is how I'm shadowing C-z:

```elisp
(bkall "C-z" 'ido-switch-buffer)
```

where bkall is

```elisp
(defun bkall (keys command)
  (mapc (lambda (m) (define-key m (kbd keys) command))
        (list evil-normal-state-map
              evil-insert-state-map
              evil-motion-state-map
              evil-emacs-state-map)))
```

**Comment by syl20bnr** [comment]
I think we could add a variable to inhibit evilication for some modes or
buffer, this should be easier for users to setup.

**Comment by ineu** [comment]
Any news on this issue? I was trying to disable auto-evilification for pdf-view-mode and neither of these work for me:
1. `(evil-set-initial-state 'pdf-view-mode 'emacs)`
2. `(push '("*.pdf" . emacs) evil-buffer-regexps)`
3. `(add-hook 'pdf-view-mode-hook #'evil-emacs-state)`

`C-z` sure helps, but is a bit annoying.
Thank you.

**Comment by dato** [comment]
I recently upgraded from master to develop, and started suffering this issue. Turns out I had hybrid mode configured thus:

```elisp
dotspacemacs-editing-style '(hybrid
                             :variables
                             hybrid-mode-enable-evilified-state nil)
```

Changing the variable to `t` solved the issue for me in git-commit-mode, and also magit-mode, which was a welcome fix for me.

Not fully understanding what it does, I still haven't noticed anything I dislike about the change. YMMV.
